providers/job.rb

```diff
@@ -116,17 +116,70 @@ def tweak_existing(job)
   # we need to specify uuid, not id to be able to update
   job.delete('id')
 
-  # rundeck gem returns project in context instead of directly project
-  # which is required to create job
-  context = job.delete('context')
-  job['project'] = context['project'] if context
+  #
+  # HACKS
+  #
+  # What follows is a sequence of hacks which should all be removed once the
+  # rundeck-ruby gem implements getting jobs in YAML, and not only in XML (as
+  # RunDeck outputs a different document hierarchy depending on the format).
+  #
+
+  # threadcount is returned as a string instead of an integer
+  unless job['dispatch'].nil? || job['dispatch']['threadcount'].nil?
```

Comment:
True, however extracting it into a library would make it seem as this is a solid thing, formally tested, etc. I wanted to make this excruciatingly obvious and painful to read so that it goes away as soon as possible.